src/vos/vos_dtx.c

```diff
@@ -717,6 +730,66 @@ vos_dtx_abort_one(struct vos_container *cont, struct daos_tx_id *dti,
 	return rc;
 }
 
+int
+vos_dtx_handle_resend(daos_handle_t coh, struct daos_tx_id *dti)
+{
+	struct vos_container	*cont;
+	struct vos_dtx_entry_df	*dtx;
+	daos_iov_t		 kiov;
+	daos_iov_t		 riov;
+	int			 rc;
+
+	cont = vos_hdl2cont(coh);
+	D_ASSERT(cont != NULL);
+
+	daos_iov_set(&kiov, dti, sizeof(*dti));
+	daos_iov_set(&riov, NULL, 0);
+	rc = dbtree_lookup(cont->vc_dtx_active_hdl, &kiov, &riov);
+	if (rc == -DER_NONEXIST)
+		rc = dbtree_lookup(cont->vc_dtx_committed_hdl, &kiov, &riov);
```

Comment:
I wonder if you should lookup in committed hdl first? since the record is updated there first? since the dtx will be put to cos cache after it get reply from all replicas, so most likely server will not get resend RPC after that.

Reply:
The DTX will be added into the active DTX table firstly, then when all replicas done the modification, the DTX will be moved to committed DTX table when it is committed some time later.

On the other hand, the active DTX table is relative smaller than the committed DTX table.

So searching the active DTX table is preferred.

Reply:
ah, I had thought active DTX table is COS cache?

Reply:
Active DTX table is in SCM, not the CoS cache which is in DRAM.